src/odemis/util/transform.py

```diff
@@ -883,337 +732,290 @@ class AffineTransform(RigidTransform):
 
     The affine transform has the following form:
 
-        x' = RSLx + t
+        x' = RLDL^Tx + t
 
-    where `R` is an orthogonal matrix, `S` is a diagonal matrix whose elements
-    represent scale factors along the coordinate axis, `L` is an upper
-    triangular matrix with all diagonal elements equal to one and a single
-    off-diagonal non-zero element, and `t` is a translation vector. To
+    where `R` is an orthogonal matrix, `L` is a lower triangular matrix with
+    all diagonal elements equal to one and a single off-diagonal non-zero
+    element, `D` is a diagonal matrix with diagonal entries `k` and `1/k` where
+    `k` is the squeeze factor, and `L^T` is the matrix transpose of `L`. To
     eliminate improper rotations (reflections) it is required that the
     determinant of `R` equals 1.
 
     Parameters
     ----------
-    matrix : (2, 2) array, optional
+    matrix : ndarray of shape (2, 2), optional
         Transformation matrix, does not include translation.
+    translation : ndarray of shape (2,), optional
+        x, y translation parameters.
+    scale : float, optional
+        Scale factor.
     rotation : float, optional
-        Rotation angle in counter-clockwise direction as radians.
-    scale : (sx, sy) as array, list, or tuple, optional
-        x, y scale factors.
+        Rotation angle in counter-clockwise direction in radians.
+    squeeze : float, optional
+        Squeeze factor. Scales the input in one axis by this factor, and in the
+        other axis by its multiplicative inverse.
     shear : float, optional
         Shear factor.
-    translation : (tx, ty) as array, list, or tuple, optional
-        x, y translation parameters.
 
     Attributes
     ----------
-    rotation_matrix
-    transformation_matrix
+    matrix : ndarray of shape (2, 2)
+    translation : ndarray of shape (2,)
+        Translation vector.
+    scale : float
+        Scale factor.
     rotation : float
-        Rotation.
-    scale : (sx, sy) as array
-        Scale factors.
+        Rotation angle in radians.
+    squeeze : float
+        Squeeze factor.
     shear : float
         Shear factor.
-    translation : (2,) array
+
+    """
+
+    def __init__(
+        self,
+        matrix: Optional[numpy.ndarray] = None,
+        translation: Optional[numpy.ndarray] = None,
+        *,
+        scale: Optional[float] = None,
+        rotation: Optional[float] = None,
+        squeeze: Optional[float] = None,
+        shear: Optional[float] = None,
+    ) -> None:
+        GeometricTransform.__init__(
```

Comment:
This was already the situation in the code before and I left it as is. They are all subclasses of each other where the actual work is in the base class. I think this is more clear, and perhaps a tiny bit faster?

Reply:
If you care about speed, don't use Python ;-) It's standard practice to use `super()` instead of explicitly indicating the name of the inheriting class. The main reason is that this avoids copy-paste error. As a bonus, it allows to handle multiple inheritance.
I didn't catch that this was already like this before. In that case, leave it as-is.